test/e2e/config_test.go

```diff
@@ -689,6 +690,121 @@ rules:
 	})
 }
 
+func TestConfigHavingRegex(t *testing.T) {
+	yaml1 := `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: game-demo
+  annotations:
+    foo1: bar1
+    foo2: bar2
+data:
+  player_initial_lives: "3"
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: game-test
+  annotations:
+    foo2: bar2
+data:
+  player_initial_lives: "3"
+`
+
+	yaml2 := `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: game-demo
+data:
+  player_initial_lives: "3"
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: game-test
+data:
+  player_initial_lives: "3"
+`
+
+	configYaml := `
+---
+apiVersion: kapp.k14s.io/v1alpha1
+kind: Config
+rebaseRules:
+  - path: [metadata, annotations, {regex: "^foo"}]
```

Comment:
Should we also add some more cases (non-happy paths)? Like empty regex?

Reply:
Empty regex will match any string so that won't make any difference here. Added a test case to check for invalid regex path like this - `path: [metadata, annotations, {regex:  }]`